wal.go

```diff
@@ -702,6 +713,10 @@ func (w *SegmentWAL) Close() error {
 	if hf := w.head(); hf != nil {
 		return errors.Wrapf(hf.Close(), "closing WAL head %s", hf.Name())
 	}
+
+	if w.dirFile != nil {
```

Comment:
Is there any case that would cause `dirFile` to be nil? I don't think so in general and just letting it panic seems ok given it would be a contract violation.